ocaml/xapi/quicktest_cbt.ml

```diff
@@ -70,6 +73,40 @@ let enable_disable_cbt_test ~session_id ~vDI =
   | Test_failed msg -> failed enable_cbt_test msg
   | e -> report_failure e enable_cbt_test
 
+(* Test data_destroy and snapshot update the necessary fields *)
+let vdi_data_destroy_test ~session_id ~vDI =
+  let data_destroy_test = make_test "Testing VDI.data_destroy" 4 in
+  try
+    start data_destroy_test;
+    debug data_destroy_test "Enabling CBT on original VDI";
+    VDI.enable_cbt ~session_id ~rpc:!rpc ~self:vDI;
+    test_assert ~test:data_destroy_test
+      (get_cbt_status ~session_id ~vDI)
+      ~msg:"VDI.enable_cbt failed";
+
+    debug data_destroy_test "Snapshotting original VDI.";
+    let newvdi = VDI.snapshot ~session_id ~rpc:!rpc ~vdi:vDI ~driver_params:[] in
+    test_assert ~test:data_destroy_test
+      (get_cbt_status ~session_id ~vDI:newvdi)
+      ~msg:"VDI.snapshot failed, cbt_enabled field didn't carry over";
+
+    debug data_destroy_test "Destroying snapshot VDI data";
+    VDI.data_destroy ~session_id ~rpc:!rpc ~self:newvdi;
+    test_compare ~test:data_destroy_test
+      (VDI.get_type ~session_id ~rpc:!rpc ~self:newvdi)
+      `cbt_metadata
+      ~msg:"VDI.data_destroy failed to update VDI.type";
+
+    let content_id_str = "/No content: this is a cbt_metadata VDI/" in
+    test_compare ~test:data_destroy_test
+      (VDI.get_other_config ~session_id ~rpc:!rpc ~self:newvdi |> List.assoc "content_id")
+      content_id_str
+      ~msg:"VDI.data_destroy failed to update VDI.content_id";
```

Comment:
It might be useful to include `content_id_str` in this message, so that people will know what went wrong in case they change the content_id string for data_destroyed VDIs.